aten/src/ATen/native/DispatchStub.h

```diff
@@ -45,8 +45,12 @@ namespace at { namespace native {
 
 enum class CPUCapability {
   DEFAULT = 0,
+#ifdef HAVE_VSX_CPU_DEFINITION
```

Comment:
This, and the other macros, seem to be named improperly. Whether or not you switch between AVX,AVX2 cpu capabilities versus VSX, shouldn't be a question of if we have VSX, but rather what *architecture* PyTorch has been built for.

I also wonder if it wouldn't be more straightforward if CPUCapability unconditionally had VSX along with AVX and AVX2, even if we were doing a good ole fashioned x86 build. The reason for this is that one of the big ongoing maintenance burdens for other architecture work like this is that problems cannot be adequately be flushed out on standard builds that most devs are using. By minimizing the amount of preprocessor macros you have, you make it less likely that these archs will bitrot.

Reply:
> I also wonder if it wouldn't be more straightforward if CPUCapability unconditionally had VSX along with AVX and AVX2, even if we were doing a good ole fashioned x86 build.

Would this cause a lot of trouble on Intel CPUs because a regular developer would have to cross build every time?

Reply:
tbh, I just copied the same style for power arch.
These are the reasons I decided to replicate the x86 style for power:
-  IBM team listed this file for modifications. so I was asked to handle dispatching as well
- A default could be used for power CPUs that do not have VSX or when the user wants to have std:: equivalent precision by setting the environment variable. 
- currently, only ppc64le are tested. so introduced VSX was not tested for Big Endian architectures. so it is good to have a fallback to DEFAULT.
-  there could be a  situation where IBM cloud virtual machine will not let to use VSX instructions. But I only encountered such situation with IBM Z architecture  cloud

Reply:
I also wanted to note that cpuinfo way handling is missing there. 
Well, it is an initial support. IBM Team probably will improve it further in the near future either on its own or with providing support.

Reply:
@VitalyFedyunin I believe dropping ifdef (in Version.cpp) is tied to this. 
If you have free time we could also discuss this PR using chat.
thanks